tests/test_execution_compute_plan.py

```diff
@@ -292,3 +292,34 @@ def test_compute_plan_circular_dependency_failure(global_execution_env):
         session.add_compute_plan(cp_spec)
 
     assert 'missing dependency among inModels IDs' in str(e.value)
+
+
+def test_execution_compute_plan_cancelled(global_execution_env):
+    factory, network = global_execution_env
+    session = network.sessions[0].copy()
+
+    dataset = session.state.datasets[0]
+
+    spec = factory.create_algo()
+    algo = session.add_algo(spec)
+
+    cp_spec = factory.create_compute_plan()
+    previous_traintuple = None
+    for data_sample_key in dataset.train_data_sample_keys:
+        previous_traintuple = cp_spec.add_traintuple(
+            algo=algo,
+            dataset=dataset,
+            data_samples=[data_sample_key],
+            in_models=[previous_traintuple] if previous_traintuple else None
+        )
+
+    cp = session.add_compute_plan()
+    session.get_traintuple(cp.traintuple_keys[0]).future().wait()
+
+    cp = session.cancel_compute_plan(cp.compute_plan_id)
+    assert cp.status == 'cancelled'
```

Comment:
we chose to use "canceled" 🇬🇧version, this will fail forever

Reply:
Updated, thanks! I used @maeldebon's new status variables for that.